sites/all/modules/reol_app_feeds/src/Helper/NodeHelper.php

```diff
@@ -0,0 +1,313 @@
+    }
+
+    return $multiple ? $values : reset($values);
+  }
+
+  /**
+   * Get text value of a field.
+   *
+   * @param object $entity
+   *   The entity.
+   * @param string $field_name
+   *   The field names.
+   * @param string|null $sub_field_name
+   *   The optional sub-field names.
+   * @param bool $multiple
+   *   If set, multiple field values will be returned as an array. Otherwise,
+   *   only the first value will be returned.
+   *
+   * @return array|mixed|null
+   *   The text field value.
+   */
+  public function getTextFieldValue($entity, $field_name, $sub_field_name = NULL, $multiple = FALSE) {
+    $values = $this->getFieldValue($entity, $field_name, $sub_field_name, TRUE);
+    $values = array_map([$this, 'getTextValue'], $values);
+
+    return $multiple ? $values : reset($values);
+  }
+
+  /**
+   * Get list of paragraphs on a node.
+   *
+   * @param object $entity
+   *   The entity.
+   *
+   * @return array|\ParagraphsItemEntity[]
+   *   The paragraphs.
+   */
+  public function getParagraphs($entity) {
+    $paragraphs = [];
+
+    $helper = new ParagraphHelper();
+
+    $fields = $helper->getParagraphFields($entity);
+    foreach ($fields as $field_name => $field) {
+      $paragraphs += $helper->loadParagraphs($entity, $field_name);
+    }
+
+    return $paragraphs;
+  }
+
+  /**
+   * Get text value.
+   *
+   * @param mixed $value
+   *   The value.
+   *
+   * @return string|null
+   *   The text value if any.
+   */
+  private function getTextValue($value) {
+    return isset($value['safe_value']) ? $value['safe_value'] : NULL;
+  }
+
+  /**
+   * Get body text value from a node.
+   *
+   * @param object $node
+   *   The node.
+   *
+   * @return string|null
+   *   The body text value if any.
+   */
+  public function getBody($node) {
+    return $this->getTextFieldValue($node, 'body', NULL, FALSE);
+  }
+
+  /**
+   * Get image url.
+   *
+   * @param mixed $value
+   *   The value.
+   * @param bool $multiple
+   *   If set, multiple values will be returned as an array. Otherwise,
+   *   only the first value will be returned.
+   * @param string|null $image_style_name
+   *   An optional image style to apply if it exists.
+   *
+   * @return string[]|string|null
+   *   The image url(s).
+   */
+  public function getImage($value, $multiple = FALSE, $image_style_name = NULL) {
+    if (NULL === $value) {
+      return NULL;
+    }
+    if (isset($value[LANGUAGE_NONE])) {
+      $value = $value[LANGUAGE_NONE];
+    }
+    if (ParagraphHelper::isAssoc($value)) {
+      $value = [$value];
+    }
+    $uris = array_column($value, 'uri');
+    if (NULL !== $image_style_name && !empty(image_style_load($image_style_name))) {
+      $urls = array_map(function ($uri) use ($image_style_name) {
+        return image_style_url($image_style_name, $uri);
+      }, $uris);
+    }
+    else {
+      $urls = array_map([$this, 'getFileUrl'], $uris);
+    }
+
+    return $multiple ? $urls : reset($urls);
```

Comment:
So one may load may images and generate image based on image style but only use the first one. Maybe check the `$multiple` before doing too much work... or do we known that it's a special case that it defaults to false or very few has more the one image uri.

Reply:
It seems that `$multiple` is never `true` – I'll remove it.